### PR TITLE
Update builds to use #749 and #756

### PR DESCRIPTION
**Pull request by joshuaboniface**
**Changes**
Updates the various build scripts and `pkg-src` files to use the updated CLI args (#749) and the new cache dir (#756). Also implements a few good ideas from the Fedora systemd/service setup into Debian.

**Issues**
N/A

Merge after #756.

### CODE REVIEW

File: deployment/debian-package-x64/pkg-src/conf/jellyfin [hunk 1]
```diff
@@ -19,10 +16,16 @@
 JELLYFIN_DATA_DIRECTORY="/var/lib/jellyfin"
 JELLYFIN_CONFIG_DIRECTORY="/etc/jellyfin"
 JELLYFIN_LOG_DIRECTORY="/var/log/jellyfin"
+JELLYFIN_CACHE_DIRECTORY="/var/cache/jellyfin"
```
Comment (JustAMan) [inline deployment/debian-package-x64/pkg-src/conf/jellyfin]: yes please, more sanity in this world!

File: deployment/debian-package-x64/pkg-src/conf/jellyfin [hunk 2]
```diff
@@ -19,10 +16,16 @@
...
+
 # Restart script for in-app server control
-JELLYFIN_RESTART_SCRIPT="/usr/lib/jellyfin/restart.sh"
-# Additional options for the binary
-JELLYFIN_ADD_OPTS=""
+JELLYFIN_RESTART_OPT="--restartpath /usr/lib/jellyfin/restart.sh"
+
+# [OPTIONAL] ffmpeg binary paths
+#JELLYFIN_FFMPEG_OPTS="-ffmpeg /usr/bin/ffmpeg -ffprobe /usr/bin/ffprobe"
```
Comment (JustAMan) [inline deployment/debian-package-x64/pkg-src/conf/jellyfin]:
```suggestion
#JELLYFIN_FFMPEG_OPTS="--ffmpeg /usr/bin/ffmpeg --ffprobe /usr/bin/ffprobe"
```
I presume single-dash options are working as a backwards compatibility thing, so I don't think we should use them in new (or edited) scripts.

File: deployment/debian-package-x64/pkg-src/postinst [hunk 3]
```diff
@@ -37,10 +38,15 @@ case "$1" in
...
     fi
+    # ensure $CACHEDATA exists
+    if [[ ! -d $CACHEDATA ]]; then
+      mkdir $CACHEDATA
+    fi
     # Ensure permissions are correct on all config directories
     chown -R jellyfin:jellyfin $PROGRAMDATA
     chown -R jellyfin:jellyfin $CONFIGDATA
     chown -R jellyfin:jellyfin $LOGDATA
+    chown -R jellyfin:jellyfin $CACHEDATA
```
Comment (JustAMan) [inline deployment/debian-package-x64/pkg-src/postinst]: Maybe condense things a little?
```suggestion
    chown -R jellyfin:jellyfin $PROGRAMDATA $CONFIGDATA $LOGDATA $CACHEDATA
```
Plus maybe `chmod o-rwx -R` to the same dirs? Privacy and stuff...

File: deployment/debian-package-x64/pkg-src/conf/jellyfin (hunk 2 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/conf/jellyfin]: No they don't any more, good catch!

File: deployment/debian-package-x64/pkg-src/postinst (hunk 3 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/postinst]: Good idea, I'm confident that wouldn't have many bad implications.

File: deployment/debian-package-x64/pkg-src/conf/jellyfin (hunk 1 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/conf/jellyfin]: :D

File: deployment/debian-package-x64/pkg-src/conf/jellyfin (hunk 2 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/conf/jellyfin]: Done.

File: deployment/debian-package-x64/pkg-src/postinst (hunk 3 above)
Comment (joshuaboniface) [reply deployment/debian-package-x64/pkg-src/postinst]: Done.

File: deployment/debian-package-x64/pkg-src/postrm [hunk 4]
```diff
@@ -55,6 +60,7 @@ case "$1" in
     # Remove anything at the default locations; catches situations where the user moved the defaults
     [[ -e /etc/jellyfin ]] && rm -rf /etc/jellyfin
     [[ -e /var/log/jellyfin ]] && rm -rf /var/log/jellyfin
+    [[ -e /var/cache/jellyfin ]] && rm -rf /var/cache/jellyfin
```
Comment (nvllsvm) [inline deployment/debian-package-x64/pkg-src/postrm]: No need to check if it exists - `rm -rf` will be successful regardless.

**Pull request merged**